yang/frr-staticd.yang

```diff
@@ -0,0 +1,150 @@
+module frr-staticd {
+  yang-version "1.1";
+  namespace "http://frrouting.org/yang/staticd";
+
+  prefix frr-staticd;
+
+  import ietf-routing-types {
+    prefix "rt-types";
+  }
+
+  import ietf-routing {
+    prefix "rt";
+  }
+
+  import ietf-ipv4-unicast-routing {
+    prefix "v4ur";
+  }
+
+  import ietf-ipv6-unicast-routing {
+    prefix "v6ur";
+  }
+
+  import frr-vrf {
+    prefix frr-vrf;
+  }
+
+  organization
+    "Free Range Routing";
+
+  contact
+    "FRR Users List: <mailto:frog@lists.frrouting.org>
+     FRR Development List: <mailto:dev@lists.frrouting.org>";
+
+  description
+    "This module defines a model for managing FRR staticd information.
+     This YANG module augments the ietf-routing with additional
+     nexthop information";
+
+  revision 2019-12-03 {
+    description
+      "Initial revision.";
+  }
+
+  grouping nexthop-common-attributes {
+    leaf table-id {
+      type uint32 {
+        range "1..4294967295";
+      }
+      description
+        "Routing table id to retrieve.";
+    }
+    leaf distance {
+      type uint8  {
+        range "1..255";
+      }
+      description
+        "Admin distance based on routing protocol.";
+    }
+    leaf mpls-label {
+      type rt-types:mpls-label;
+      description
+        "Nexthop's MPLS label stack.";
+    }
+    leaf nexthop-vrf {
+      type string {
+        length "1..36";
+      }
+      description
+        "The nexthop vrf name, if different from the route.";
+    }
+  }
+
+  grouping frr-staticd-nexthop {
+
+    choice next-hop-type {
+      case  gateway {
+        leaf gw-interface {
+          type string {
+            length "1..16";
+          }
+          description
+            "Name of the outgoing interface.";
+        }
+        leaf is-onlink {
+          type empty;
+            description
+              "Nexthop is directly connected.";
+        }
+      }
+      case ifname {
+        leaf outgoing-interface {
+          type string {
+            length "1..16";
+          }
+          description
+            "Name of the outgoing interface.";
+        }
+      }
+      case Null0 {
+        leaf Null0 {
```

Comment:
This is a bit of a special case, since `Null0` is treated as a "special" interface name. IMO this doesn't need to be represented in the model at all, it's really an anachronism from other vendors, and it's essentially an alias for the `blackhole` attribute.